xbmc/pvr/windows/GUIWindowPVRGuide.cpp

```diff
@@ -592,28 +592,93 @@ bool CGUIWindowPVRGuideBase::OnMessage(CGUIMessage& message)
 
 bool CGUIWindowPVRGuideBase::OnContextButton(int itemNumber, CONTEXT_BUTTON button)
 {
-  switch (button)
-  {
-    case CONTEXT_BUTTON_BEGIN:
-      return OnContextButtonBegin();
+  if (OnContextButtonNavigate(button))
+    return true;
 
-    case CONTEXT_BUTTON_NOW:
-      return OnContextButtonNow();
+  if (itemNumber < 0 || itemNumber >= m_vecItems->Size())
+    return false;
+
+  return CGUIMediaWindow::OnContextButton(itemNumber, button);
+}
 
-    case CONTEXT_BUTTON_DATE:
-      return OnContextButtonDate();
+namespace
+{
 
-    case CONTEXT_BUTTON_END:
-      return OnContextButtonEnd();
+template<typename A>
+class CContextMenuFunctions : public CContextButtons
+{
+public:
+  explicit CContextMenuFunctions(A* instance) : m_instance(instance) {}
 
-    default:
-      break;
+  void Add(bool (A::*function)(), unsigned int resId)
+  {
+    CContextButtons::Add(size(), resId);
+    m_functions.emplace_back(std::bind(function, m_instance));
   }
 
-  if (itemNumber < 0 || itemNumber >= m_vecItems->Size())
-    return false;
+  bool Call(int idx)
+  {
+    if (idx < 0 || idx >= static_cast<int>(m_functions.size()))
+      return false;
 
-  return CGUIMediaWindow::OnContextButton(itemNumber, button);
+    return m_functions[idx]();
+  }
+
+private:
+  A* m_instance = nullptr;
+  std::vector<std::function<bool()>> m_functions;
+};
+
+} // unnamed namespace
+
+bool CGUIWindowPVRGuideBase::OnContextButtonNavigate(CONTEXT_BUTTON button)
+{
+  bool bReturn = false;
+
+  if (button == CONTEXT_BUTTON_NAVIGATE)
+  {
+    if (g_SkinInfo->HasSkinFile("DialogPVRGuideControls.xml"))
+    {
+      // use controls dialog
+      CGUIDialog* dialog = CServiceBroker::GetGUI()->GetWindowManager().GetDialog(WINDOW_DIALOG_PVR_GUIDE_CONTROLS);
+      if (dialog && !dialog->IsDialogRunning())
+      {
+        dialog->Open();
+      }
+    }
+    else
```

Comment:
If this is for v19, does it make sense to keep the fallback menu at all?

Reply:
I thought it would be nice not to enforce every skin to implement the new dialog...